packages/ui/src/proposals/modals/AddNewProposal/components/SpecificParameters/RuntimeUpgrade.tsx

```diff
@@ -26,9 +26,29 @@ const MAX_FILE_SIZE = 3 * 1024 * 1024
 
 export const RuntimeUpgrade = ({ setRuntime }: RuntimeUpgradeProps) => {
   const onDrop = useCallback(async (acceptedFiles: File[]) => {
-    const file = acceptedFiles[0]
-    setRuntime(await file.arrayBuffer())
+    if (acceptedFiles.length) {
+      setRuntime(await acceptedFiles[0].arrayBuffer())
+    }
   }, [])
+  const [fileValidation, setFileValidation] = useState<{ file: File; isValid: boolean } | undefined>()
```

Comment:
I think that another variable is not needed here. Using [the react-dropzone way](https://react-dropzone.js.org/#section-extending-dropzone) allows using file objects with custom prop:

```javascript
interface ValidatedFile extends File {
  isValidWASM?: boolean
}
```